// react-native-animated-math/example/App.js
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Animated,
  Easing,
  Dimensions,
  Image
} from 'react-native';

import AnimatedMath from 'react-native-animated-math';

const { width, height } = Dimensions.get('window');

const colors = {
  green: 'rgb(75, 167, 118)'
};

class Rotator extends Component {
  state = {
    distance: new Animated.Value(100),
    angle: new Animated.Value(0),
  }

  componentDidMount() {
    Animated.parallel([

      Animated.timing(this.state.angle, {
        toValue: 4*Math.PI,
        duration: 20000,
        easing: Easing.linear
      })
    ]).start();
  }

  render() {
    let {angle} = this.state;
    distance = 130;

    return (
      <Animated.View style={[styles.little, {
        transform: [
          {translateX: Animated.multiply(AnimatedMath.sinus(angle), this.state.distance)},
          {translateY: Animated.multiply(AnimatedMath.cosinus(angle), this.state.distance)},
        ],
        opacity: 1
      }]}/>
    );
  }
}

export default class App extends Component {
  state = {
    distance: new Animated.Value(0),
    angle: new Animated.Value(0)
  }

  onComponentDidMount() {
    this.animate();
  }

  animate() {
    Animated.timing(this.state.angle, {
      toValue: 2*Math.PI,
      duration: 1000,
      easing: Easing.linear
    });
  }

  render() {
    return (
      <View style={styles.container}>

        <Rotator />

        <TouchableOpacity style={{position: 'absolute', left: 5, bottom: 5}}>
          <Text style={{backgroundColor: 'transparent'}} onPress={this.reset}>reset</Text>
        </TouchableOpacity>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  little: {
    position: 'absolute',
    bottom: 50+height/2,
    width: 20,
    height: 20,
    borderRadius: 10,
    backgroundColor: colors.green,
  }
});
